ArchUnitNET/Loader/ArchBuilder.cs

```diff
@@ -57,22 +57,21 @@ public void LoadTypesForModule(ModuleDefinition module, string namespaceFilter)
         {
             _architectureCacheKey.Add(module.Name, namespaceFilter);
 
-            ICollection<TypeDefinition> types;
+            var types = module.Types.First().FullName.Contains("<Module>")
+                ? module.Types.Skip(1).ToList()
+                : module.Types.ToList();
 
-            if (module.Types.First().FullName.Contains("<Module>"))
+            var nestedTypes = types;
+            var depth = 0;
+            do
             {
-                types = module.Types.Skip(1).ToList();
-            }
-            else
-            {
-                types = module.Types;
-            }
-
+                depth++;
+                nestedTypes = nestedTypes.SelectMany(typeDefinition =>
+                    typeDefinition.NestedTypes.Where(type => !type.IsCompilerGenerated())).ToList();
+                types.AddRange(nestedTypes);
+            } while (nestedTypes.Any() && depth < 10);
```

Comment:
Why do we need to limit the depth? If we really need to, I think this limit should be higher and we should at least output a warning. But IMHO we don't need a limit.

Reply:
We don't really need it so I removed it.